### PR TITLE
Bugfix/handle datatypes for inputs

### PR DESCRIPTION
**Pull request by tjellicoe-tpzuk**
## Update Type Cast to be Boolean
- Corrected type cast for boolean data types
- Also pulled in latest changes from upstream repository

### CODE REVIEW

File: zoo_calrissian_runner/__init__.py [hunk 1]
```diff
@@ -221,19 +223,38 @@ def get_input_value(self, key):
...
         """Returns a list with the input parameters keys"""
         res={}
+        hasVal=False
         for key, value in self.inputs.items():
             if "dataType" in value:
-                match value["dataType"]:
-                    case w if w in ["double","float"]:
+                if isinstance(value["dataType"],list):
+                    # How should we pass array for an input?
+                    import json
```
Comment (apalmer-tpzuk) [inline zoo_calrissian_runner/__init__.py]: json import seems not needed?

File: zoo_calrissian_runner/__init__.py [hunk 2]
```diff
@@ -221,19 +223,38 @@ def get_input_value(self, key):
     def get_processing_parameters(self):
         """Returns a list with the input parameters keys"""
         res={}
+        hasVal=False
```
Comment (apalmer-tpzuk) [inline zoo_calrissian_runner/__init__.py]: hasVal not used

File: zoo_calrissian_runner/__init__.py [hunk 3]
```diff
@@ -221,19 +223,38 @@ def get_input_value(self, key):
...
                         res[key]=float(value["value"])
-                    case "integer":
+                    elif value["dataType"] == "integer":
                         res[key]=int(value["value"])
-                    case "boolean":
+                    elif value["dataType"] == "boolean":
                         res[key]=bool(value["value"])
-                    case _:
+                    else:
                         res[key]=value["value"]
```
Comment (apalmer-tpzuk) [inline zoo_calrissian_runner/__init__.py]: Any particular reason to switch from case to if/else? and any logical reason why list is separated from the other if/elif chain?

File: zoo_calrissian_runner/__init__.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import inspect
 import os
+import sys
```
Comment (apalmer-tpzuk) [inline zoo_calrissian_runner/__init__.py]: sys not used

File: zoo_calrissian_runner/__init__.py (hunk 2 above)
Comment (tjellicoe-tpzuk) [reply zoo_calrissian_runner/__init__.py]: removed

File: zoo_calrissian_runner/__init__.py (hunk 1 above)
Comment (tjellicoe-tpzuk) [reply zoo_calrissian_runner/__init__.py]: removed

File: zoo_calrissian_runner/__init__.py (hunk 4 above)
Comment (tjellicoe-tpzuk) [reply zoo_calrissian_runner/__init__.py]: removed

File: zoo_calrissian_runner/__init__.py (hunk 3 above)
Comment (tjellicoe-tpzuk) [reply zoo_calrissian_runner/__init__.py]: good point, copied from parent repo, I will update this to be case statement.
I expect `list` is handled separately as it means each item in the array could be a different datatype, so will leave this as is

**Review by apalmer-tpzuk — APPROVED** [review]

File: zoo_calrissian_runner/__init__.py (hunk 3 above)
Comment (tjellicoe-tpzuk) [reply zoo_calrissian_runner/__init__.py]: This causes an issue due to the Python version used by the ZOO Project build, so for now I have set this back to if...else

**Pull request merged**